### DISCUSSION TITLE
Issues generating from openapi published spec

### DISCUSSION THREAD

**Discussion by lemonprogis** [weather-gov/api#594] [Q&A]
Hello! 

Firstly, thanks for the availability of a public API. I appreciate it. 

Second, I'm encountering the following when trying to validate/generate a client off of the openapi sepcification provided from api.weather.gov/openapi.json 

I get the following 3 errors from the specification. Is there a place we can fix this from the community or is it just a versioning issue? thanks :)
<img width="1361" alt="image" src="https://user-images.githubusercontent.com/1834551/204151963-98faaa65-a8a5-4d78-8607-2d87b20f4aeb.png">

**Comment by lemonprogis** [comment]
#595 I've made an attempt for the spec fixes. it passes validation and generates correctly.

**Reply by dwhitemv25** [reply]
We'd discussed the codegen issues previously in https://github.com/weather-gov/api/discussions/559. The additionalProperties hack "works," but only because the specification is incomplete. Allowing an empty schema property needs a much newer version of OpenAPI (and related validators) than the spec is marked for. For the record, what version of openapi-generator were you using?

**Reply by lemonprogis** [reply]
Hey, thanks for the reply! 

Using 6.2.1 from CLI version @openapitools/openapi-generator-cli 2.5.2. So that JSON-LTD is what is incomplete there? I was able to swap like the PR has done and get my client generated and seems to be working so I can just keep this local to my application, but wanted to pass it along.
